File: src/main/java/org/bk/graphql/web/dto/AuthorDto.java
package org.bk.graphql.web.dto;

import org.bk.graphql.domain.Author;

public record AuthorDto(String id, String name, int version) {
    public static AuthorDto map(Author author) {
        return new AuthorDto(author.id().id(), author.name(), author.version());
    }
}

File: src/main/kotlin/org/bk/graphql/web/dto/AuthorDto.kt
package org.bk.graphql.web.dto

import org.bk.graphql.domain.Author

data class AuthorDto(
    val id: String,
    val name: String,
    val version: Int

) {
    companion object {
        fun map(author: Author): AuthorDto {
            return AuthorDto(author.id.id, author.name, author.version)
        }
    }
}


